Sorting/RadixSort.cpp

```diff
@@ -0,0 +1,113 @@
+/*
+    Radix sort
+    ----------
+    An efficient, key-index-based, comparision based sorting algorithm.
+    It sorts based on the keys obtained on each index of the numbers to be sorteed.
+*/
+
+#include <iostream>
+#include <vector>
+#include <array>
+#include "SortingUtils.hpp"
+
+using namespace std;
+
+// The setFactors function sets the factors which depend on whether the sorting is needed to be done in ascending or descending order
+
+/*
+    count_sort
+    ----------
+    Performs Count Sort on the elements passed by the radix_sort function
+
+    Time complexity
+    ---------------
+    O(N + B), where N is the number of keys and B is the base for representing the numbers(10 in case of decimal system)
+
+    Space complexity
+    ----------------
+    O(N + B), where N is the number of keys and B is the base for representing the numbers(10 in case of decimal system)
+    This space is used for count sorting each key and temporarily creating a sorted array.
+*/
+void count_sort(vector<int>& values, const int extractor, const bool to_show_state, const int mult_factor, const int add_factor) {
+    array<int, 10> counter{0};
+    vector<int> output(values.size());
+    for(int i = 0; i < values.size(); i++)
+        counter[add_factor + (mult_factor * ((values[i]/extractor) % 10))]++;
```

Comment:
Using a range-based for loop here will work as intended, so please add it back.
On line 38 however, since you are iterating the container in a reverse order, you cannot use a range-based for loop as it iterates from the beginning until the end.